### ISSUE TITLE
How to save weights and models

### ISSUE THREAD

**Issue by mengxiabing** [matterport/Mask_RCNN#329]
How to save weights and models

**Comment by fastlater** [comment]
Weights are saved after each training epoch inside logs folder (.h5 files).
You can check https://keras.io/getting-started/faq/ (Saving/loading only a model's weights section) for further info.

**Comment by mengxiabing** [comment]
@fastlater Thank you. I need a model of the network structure, not just a weight file. How can I export it?

**Comment by fastlater** [comment]
@mengxiabing  in the same link, you will find how to save the architecture of the model.

In model.py, you can see that they used keras so you need to follow the instructions from keras about how to save the model. I am not sure if there is a method inside this code to save the model architecture.

**Comment by CMCDragonkai** [comment]
It appears the model cannot be easily saved. I encountered this error when trying to save MaskRCNN model: https://github.com/keras-team/keras/issues/10140

**Comment by lovebabychen** [comment]
> How to save weights and models

Now,is the question you face resolved? if “yes” ，please tell me how to solve it
thanks

**Comment by shen338** [comment]
Run into the same thing, anyone knows?

**Comment by K-M-Ibrahim-Khalilullah** [comment]
I am facing the same problem. Anyone know about it, please help me..

**Comment by NayanDharviya** [comment]
how to save mask rcnn cfg and weights file to serve the model using tensorflow serving?

**Comment by avnerst** [comment]
> Weights are saved after each training epoch inside logs folder (.h5 files). You can check https://keras.io/getting-started/faq/ (Saving/loading only a model's weights section) for further info.

the saved weights of the model don't appear for me after each epoch, but only after some selected ones (only epochs 1 and 3 for example), has this happened to you? does anybody know how to fix it?
